### PR TITLE
Fix OOM on struct cloning

### PR DESCRIPTION
**Pull request by roman-khimov**
See neo-project/neo#2534.

### CODE REVIEW

File: pkg/vm/vm.go [hunk 1]
```diff
@@ -1370,12 +1373,19 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 			src := t.Value().([]stackitem.Item)
 			arr = make([]stackitem.Item, len(src))
 			for i := range src {
-				arr[i] = cloneIfStruct(src[i])
+				arr[i], err = cloneIfStruct(src[i])
+				if err != nil {
+					panic(err)
+				}
```
Comment (fyrchik) [inline pkg/vm/vm.go:1379]: Can we hide it inside `cloneIfStruct`? It is private and used under `execute` protection.

File: pkg/vm/stackitem/item.go [hunk 2]
```diff
@@ -298,17 +298,30 @@ func (i *Struct) Convert(typ Type) (Item, error) {
...
 		switch t := i.value[j].(type) {
 		case *Struct:
-			ret.value[j] = t.Clone()
+			var err error
+
+			ret.value[j], err = t.clone(limit)
+			if err != nil {
+				return nil, err
+			}
+			*limit--
```
Comment (fyrchik) [inline pkg/vm/stackitem/item.go:316]: I think we can also update limit for primitive items.

Comment (roman-khimov) [reply pkg/vm/stackitem/item.go:316]: Yeah, it's mentioned in the commit message, but I've specifically decided to not complicate things with it.

**Review by AnnaShaleva — APPROVED** [review]

File: pkg/vm/vm.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/vm/vm.go]: Fixed,

**Review by fyrchik — APPROVED** [review]

**Pull request merged**